source/common/config/grpc_mux_impl.cc

```diff
@@ -149,16 +145,28 @@ void GrpcMuxImpl::onDiscoveryResponse(
     return;
   }
 
-  if (apiStateFor(type_url).watches_.empty()) {
+  ApiState& api_state = apiStateFor(type_url);
+
+  if (message->has_control_plane()) {
+    control_plane_stats.identifier_.set(message->control_plane().identifier());
+  }
+
+  if (message->control_plane().identifier() != api_state.control_plane_identifier_) {
+    api_state.control_plane_identifier_ = message->control_plane().identifier();
+    ENVOY_LOG(info, "Receiving gRPC updates for {} from {}", type_url,
```

Comment:
Do we always want to know about control plane id change? Perhaps `debug` level would work here?

Reply:
Done.